sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/AnalyticsTransmissionTarget.java

```diff
@@ -80,4 +147,29 @@ public synchronized AnalyticsTransmissionTarget getTransmissionTarget(String tra
     String getTransmissionTargetToken() {
         return mTransmissionTargetToken;
     }
+
+    @NonNull
+    private String getEnabledPreferenceKey() {
+        return Analytics.getInstance().getEnabledPreferenceKeyPrefix() + mTransmissionTargetToken.split("-")[0];
+    }
+
+    @WorkerThread
+    private boolean isEnabledInStorage() {
+        return StorageHelper.PreferencesStorage.getBoolean(getEnabledPreferenceKey(), true);
+    }
+
+    @WorkerThread
+    private boolean areAncestorsEnabled() {
+        for (AnalyticsTransmissionTarget target = mParentTarget; target != null; target = target.mParentTarget) {
```

Comment:
@clpolet and I talked about this and you shouldn't have to check anything other than the immediate parent.  If the immediate parent is disabled then we should not allow children to be enabled.  If the immediate parent is enabled, that means that nothing in the chain above it could be disabled, because otherwise the immediate parent would be disabled.

This logic should be able to be simplified this way.